### PR TITLE
Un-comment and fixes all snippets

### PR DESCRIPTION
**Pull request by yasserzamani**
# Changes
+ Supports snippets with base path `struts2/` and `struts2-tags/`
+ Caches fetched contents in files
+ Prints to `STDERR` on snippet loading failure but continue to next
+ Deletes `~~~~~~` before and after snippet (a RegEx replace)

# Test results

All snippets successfully loaded and generated except all `dojo` snippets and followings:

```text
START SNIPPET: javadoc not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=core/src/main/java/org/apache/struts2/components/DateTextField.java;hb=HEAD
START SNIPPET: example not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=core/src/main/java/org/apache/struts2/components/DateTextField.java;hb=HEAD
START SNIPPET: javadoc not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=core/src/main/java/org/apache/struts2/components/Div.java;hb=HEAD
START SNIPPET: javadoc not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=core/src/main/java/org/apache/struts2/components/DateTextField.java;hb=HEAD
START SNIPPET: example not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=core/src/main/java/org/apache/struts2/components/DateTextField.java;hb=HEAD
START SNIPPET: javadoc not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=core/src/main/java/org/apache/struts2/components/Div.java;hb=HEAD
START SNIPPET: description not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=core/src/main/java/org/apache/struts2/dispatcher/PostbackResult.java;hb=HEAD
START SNIPPET: params not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=core/src/main/java/org/apache/struts2/dispatcher/PostbackResult.java;hb=HEAD
START SNIPPET: example not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=core/src/main/java/org/apache/struts2/dispatcher/PostbackResult.java;hb=HEAD
START SNIPPET: visitorValidatorsExample not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=apps/showcase/src/main/resources/org/apache/struts2/showcase/validation/VisitorValidatorsExampleAction-submitVisitorValidatorsExamples-validation.xml;hb=HEAD
START SNIPPET: description not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=core/src/main/java/org/apache/struts2/dispatcher/VelocityResult.java;hb=HEAD
START SNIPPET: params not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=core/src/main/java/org/apache/struts2/dispatcher/VelocityResult.java;hb=HEAD
START SNIPPET: example not found in https://gitbox.apache.org/repos/asf?p=struts.git;a=blob_plain;f=core/src/main/java/org/apache/struts2/dispatcher/VelocityResult.java;hb=HEAD
```

### CODE REVIEW

**Comment by lukaszlenart** [comment]
I think there is still one issue, when a snippet contains `lang=html` it should be inlined into a page not escaped as a code. Except that, nice work 👍

File: source/_plugins/snippet.rb [hunk 1]
```diff
@@ -30,9 +27,10 @@ def render(context)
       if @content
         snippet = @content
         if @id
-          snippet = snippet[/#{Regexp.escape("<!-- START SNIPPET: " + @id + " -->")}[^\n]*\n(.*?)\n[^\n]*#{Regexp.escape("<!-- END SNIPPET: " + @id + " -->")}/m, 1]
+          snippet = snippet[/#{Regexp.escape("START SNIPPET: " + @id)}[^\n]*\n(.*?)\n[^\n]*#{Regexp.escape("END SNIPPET: " + @id)}/m, 1]
```
Comment (lukaszlenart) [inline source/_plugins/snippet.rb]: One more thing, right now the whole snippet will be included, also we comments:
```
<!-- START SNIPPET: method-example -->
<s:form action="baz">
    <s:textfield label="Enter your name" name="person.name"/>
    <s:submit value="Create person"/>
    <s:submit method="anotherMethod" value="Cancel"/>
</s:form>
<!-- END SNIPPET: method-example -->
```

http://struts.apache.org/core-developers/action-mapper.html

Comment (yasserzamani) [reply source/_plugins/snippet.rb]: Thank you! I examined and saw it's not a snippet tag there at all; It's hard-coded raw text block :P

Comment (lukaszlenart) [reply source/_plugins/snippet.rb]: LoL! That's good :)

**Pull request merged**